src/dos/drive_iso.cpp

```diff
@@ -503,7 +503,11 @@ int isoDrive :: readDirEntry(isoDirEntry *de, Bit8u *data) {
 	if (dotpos!=NULL) {
 		if (strlen(dotpos)>4) dotpos[4]=0;
 		if (dotpos-(char*)de->ident>8) {
-			strcpy(reinterpret_cast<char *>(&de->ident[8]), dotpos);
+			constexpr int pos = 8;
+			const auto maxlen = sizeof(de->ident) - pos;
+			const auto sub_ident = reinterpret_cast<char *>(
+			        de->ident + pos);
```

Comment:
Not sure, but perhaps this way we could avoid reinterpret cast?
```suggestion
			const auto sub_ident = &de->ident[pos];
```

Reply:
Will give that a try and report back.

Reply:
error: no matching function for call to 'snprintf' `snprintf(sub_ident, maxlen, "%s", dotpos);`

``` text
/usr/include/stdio.h:354:12: note: candidate function not viable: no known conversion from 'unsigned char *const' to 'char *__restrict' for 1st argument extern int snprintf (char *__restrict __s, size_t __maxlen
```

The ident array is a `uint8_t[]`, so yeah; only option it to reinterpret the pointer type.